app/src/main/java/com/example/fideicomisoapproverring/FindEscrowActivity.kt

```
@@ -98,4 +140,130 @@ class FindEscrowActivity : AppCompatActivity() {
+    private fun parseBalance(response: String?): String {
+        return try {
+            val json = JSONObject(response ?: return "0")
+            val balances = json.getJSONArray("balances")
+            for (i in 0 until balances.length()) {
+                val balance = balances.getJSONObject(i)
+                if (balance.getString("asset_type") == "native") {
+                    return balance.getString("balance")
+                }
+            }
+            "0"
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)